core/appeal/service.go

```diff
@@ -770,15 +770,15 @@ func (s *Service) handleAppealRequirements(ctx context.Context, a *domain.Appeal
 					additionalAppeal.PolicyID = aa.Policy.ID
 					additionalAppeal.PolicyVersion = uint(aa.Policy.Version)
 				}
-				additionalAppeals = append(additionalAppeals, additionalAppeal)
+				if err := s.Create(ctx, []*domain.Appeal{additionalAppeal}); err != nil {
+					if errors.Is(err, ErrAppealDuplicate) {
+						continue
+					}
+					return fmt.Errorf("creating additional appeals: %v", err)
```

Comment:
```suggestion
					return fmt.Errorf("creating additional appeals: %w", err)
```
better to use `%w` to wrap the error